bin/p2-inspect/main.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/armon/consul-api"
+	"github.com/square/p2/pkg/intent"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/reality"
+	"gopkg.in/alecthomas/kingpin.v1"
+)
+
+var (
+	consulUrl = kingpin.Flag("consul", "The hostname and port of a consul agent in the p2 cluster. Defaults to 0.0.0.0:8500.").String()
+	nodeName  = kingpin.Flag("node", "The node to inspect. By default, all nodes are shown.").String()
+	podId     = kingpin.Flag("pod", "The pod manifest ID to inspect. By default, all pods are shown.").String()
+)
+
+type NodePodStatus struct {
+	NodeName           string `json:"-"`
+	PodId              string `json:"-"`
+	IntentManifestSHA  string
```

Comment:
I'd suggest converting these fields to `snake_case` in the json marshal. That's consistent with the format of fields in the pod manifest.